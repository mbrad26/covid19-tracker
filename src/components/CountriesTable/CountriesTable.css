#table {
  overflow: scroll;
  height: 520px;
  border-radius: 7px 7px 0 0;
  border-bottom: 3px solid #009879;
  font-size: 0.9em;
  margin-bottom: 10px;
}

.table thead tr {
  background-color: #009879;
  color: #ffffff;
  height: 55px;
}

.table tbody td, .table thead th {
  padding: 5px;
  text-align: center;
  vertical-align: middle;
}

.table tbody {
  background-color: #ffffff;
}

#cases {
  margin-right: 0px;
}

#recovered {
  margin-left: 0px;
  padding-right: 10px;
}

img {
  width: 50px;
  height: 35px;
}

.table thead #form {
  width: 174px;
  margin: auto;
  overflow: hidden;
  border-radius: 8px;
}

#cases:hover,
#deaths:hover,
#recovered:hover {
  cursor: pointer;
}

.table thead button {
  background-color: inherit;
  color: #ffffff;
  font-weight: bold;
}

button,
button:hover {
  border-style: hidden;
  border: none;
}

button:focus {
  outline: none;
  font-weight: bold;
}
