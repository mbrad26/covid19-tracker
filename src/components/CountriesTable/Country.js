import React from 'react';
import { useDispatch } from 'react-redux';
import { Image } from 'react-bootstrap';

import { doDataSuccess } from '../../actions/globalData';
import { doUpdateEndPoint, doHistoricalCountryDataLoading } from '../../actions/historicalData';

const Country = ({ country }) => {
  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(doDataSuccess(country));
    dispatch(doUpdateEndPoint(country.country));
    dispatch(doHistoricalCountryDataLoading());
  };

  return (
    <tr>
      <td><Image
            src={country.countryInfo.flag}
            alt={`${country.country} flag`}
            id='flag'
          />
      </td>
      <td>
        <button
          onClick={handleClick}
        >
          {country.country}
        </button>
      </td>
      <td>{country.cases}</td>
      <td>{country.deaths}</td>
      <td>{country.recovered}</td>
    </tr>
  );
};

export default Country;
