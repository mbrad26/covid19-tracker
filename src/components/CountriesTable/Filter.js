import React from 'react';

const Filter = ({
  query,
  reset,
  setQuery,
  globalData,
  dataLoading,
  dataSuccess,
  zoomOnCountry,
  updateEndPoint,
  filteredCountries,
  historicalDataLoading,
  historicalCountryDataLoading,
  }) => {

  const handleChange = event => {
    setQuery(event.target.value);
  };

  const handleSubmit = event => {
    event.preventDefault();
    if (filteredCountries.length === 1) {
      dataSuccess(filteredCountries[0]);
      zoomOnCountry(filteredCountries[0]);
      updateEndPoint(filteredCountries[0].country);
      historicalCountryDataLoading();
    } else {
      reset();
      dataLoading();
      dataSuccess(globalData);
      historicalDataLoading();
    }
    setQuery('');
  };

  console.log('FILTER');

  return (
    <div id='form'>
      <form onSubmit={handleSubmit}>
        <input
          type='text'
          placeholder='Search ...'
          value={query}
          onChange={handleChange}
          autoFocus
        />
      </form>
    </div>
  )
}

export default Filter;
