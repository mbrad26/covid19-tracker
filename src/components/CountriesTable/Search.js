import React from 'react';

const Search = () => (
  <div id='form'>
    <form>
      <input
        placeholder='Search for ...'
      />
    </form>
  </div>
)

export default Search;
