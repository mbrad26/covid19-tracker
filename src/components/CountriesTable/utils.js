export const filterData = (query, data) =>
        data.filter(country =>
          country.country.toLowerCase().includes(query.toLowerCase()));
