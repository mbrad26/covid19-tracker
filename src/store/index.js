import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  applyMiddleware(saga, logger)
);

// console.log('STORE: ', store.getState());

saga.run(rootSaga);

export default store;
