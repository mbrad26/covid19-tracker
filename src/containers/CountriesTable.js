import { connect } from 'react-redux';

import CountriesTable from '../components/CountriesTable/CountriesTable';
import { getIsLoadingStatus, getData, getIsErrorStatus } from '../selectors/data';
import { doCountriesDataLoading, doCountriesDataSort } from '../actions/countriesData';

const mapStateToProps = ({ countriesDataState }) => ({
  data: getData(countriesDataState),
  isError: getIsErrorStatus(countriesDataState),
  isLoading: getIsLoadingStatus(countriesDataState),
});

const mapDispatchToProps = dispatch => ({
  loadingData: () => dispatch(doCountriesDataLoading()),
  onSortData: (key, sort) => dispatch(doCountriesDataSort(key, sort)),
});

export default connect(mapStateToProps, mapDispatchToProps)(CountriesTable);
