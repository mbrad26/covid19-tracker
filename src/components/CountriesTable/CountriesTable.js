import React, { useState, useEffect } from 'react';
import { Table } from 'react-bootstrap';

import Country from './Country';
import Filter from './Filter';
import './CountriesTable.css';

const CountriesTable = ({ data, loadingData, onSortData }) => {
  const [sort, setSort] = useState('asc');
  const [query, setQuery] = useState('');

  console.log('CountriesTable');

  useEffect(() => {
    loadingData()
  }, [loadingData]);

  const handleClick = event => {
    sort === 'asc' ? setSort('desc') : setSort('asc');
    onSortData(event.target.getAttribute('id'), sort);
  };

  const filterData = data =>
          data.filter(country =>
            country.country.toLowerCase().includes(query.toLowerCase()));

  return (
    <div className='mt-3 shadow' id='table'>
      <Table className='shadow'>
        <thead>
          <tr>
            <th>Flag</th>
            <th>
              <Filter
                query={query}
                setQuery={setQuery}
              />
            </th>
            <th>
              <button onClick={handleClick} id='cases'>
                Cases
              </button>
            </th>
            <th>
              <button onClick={handleClick} id='deaths'>
                Deaths
              </button>
            </th>
            <th>
              <button onClick={handleClick} id='recovered'>
                Recovered
              </button>
            </th>
          </tr>
        </thead>
        <tbody>
        {filterData(data).map(country =>
          <Country key={country.country} country={country} />
        )}
        </tbody>
      </Table>
    </div>
  )
}

export default CountriesTable;
