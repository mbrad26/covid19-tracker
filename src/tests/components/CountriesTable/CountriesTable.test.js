import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import store from '../../../store';
import CountriesTable from '../../../components/CountriesTable/CountriesTable';

describe('CountriesTable', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <Provider store={store} >
        <CountriesTable
          data={[]}
          isError={false}
          isLoading={false}
          onSortData={jest.fn()}
          loadingData={jest.fn()}
        />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
