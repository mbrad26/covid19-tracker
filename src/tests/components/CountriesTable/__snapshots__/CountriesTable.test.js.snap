// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesTable renders snapshot 1`] = `
<div
  id="table-container"
>
  <table
    class="table table-sm"
  >
    <thead>
      <tr>
        <th>
          Flag
        </th>
        <th>
          <div
            id="form"
          >
            <form>
              <input
                placeholder="Search ..."
                type="text"
                value=""
              />
            </form>
          </div>
        </th>
        <th>
          <button
            id="cases"
          >
            Cases
          </button>
        </th>
        <th>
          <button
            id="deaths"
          >
            Deaths
          </button>
        </th>
        <th>
          <button
            id="recovered"
          >
            Recovered
          </button>
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
